### ISSUE TITLE
Docker restart while project is running results in port errors on `ddev start` due to macOS docker bug

### ISSUE THREAD

**Issue by rfay** [drud/ddev#2981]
Symptom: `Failed to start d9: Unable to listen on required ports, port 443 is already in use,
Troubleshooting suggestions at https://ddev.readthedocs.io/en/stable/users/troubleshooting/#unable-listen`

**Run a Diagnostic and Paste Link Here**

test_ddev.sh output: https://gist.github.com/rfay/f29c71dcde73c59cefd5257b51b37225

**Describe the bug**

If you leave at least one project running (which leaves ddev-router running) in v1.17+, and restart docker or restart the computer, then start a project (the same one or a different one), you get

```
ERROR: for ddev-router  Cannot start service ddev-router: Ports are not available: port is already allocated.
```

This appears to be a docker bug with restarting ddev-router.

Update: This is in fact a newer bug in Docker Desktop for Mac, https://github.com/docker/for-mac/issues/5649

**To Reproduce**

1. Start a project
2. Restart docker
3. Start the project again.

In my experience this can always be fixed with `ddev poweroff` and (often) a docker restart. It's also easily prevented by a `ddev poweroff` before computer shutdown or docker restart.

**Expected behavior**

It should just start the project.

**Version and configuration information (please complete the following information):**

- Host computer OS and Version: macOS M1 Big Sur
- Docker Desktop version: 3.3.1
- ddev version information : v1.17.2

**Additional Context**

This seems to be mostly caused by a related Docker Desktop problem on macOS amd64 and arm64: https://github.com/docker/for-mac/issues/5649

A perhaps related failure was reported in [TYPO3 #ddev slack](https://typo3.slack.com/archives/C8TRNQ601/p1619766793213500?thread_ts=1619766682.213200&cid=C8TRNQ601)

https://gist.github.com/rfay/cb0228078eb928c00560e9b133bf352c

**Comment by rfay** [comment]
Reopening this, because the real problem is upstream, and we want people to be able to find it.

**Comment by rfay** [comment]
To prevent this docker problem, avoid rebooting or powering off or restarting docker while any ddev project is running.

To fix things when you have this problem:
`ddev poweroff` and then restart docker.

**Comment by JMartinRootStack** [comment]
If you are facing this issue, consider enabling the option "**Allow the default Docker socket to be used (requires password)**" under "**Advanced**" in the "**Settings**" screen. First, `poweroff` ddev, change config, and explicitly select "**restart**" option from the dropdown menú of docker desktop. Do not just rely on it's "restart" after the configuration has been updated.

After I reproduced these steps, I stopped having the issue. I have Docker Desktop 4.20.1 and Ddev 1.21.6
